### PR TITLE
PubMatic Adapter:  add support for FloCID

### PR DESCRIPTION
**Pull request by pm-shashank-jain**
## Type of change
- [x] Feature

## Description of change
Reading of FloC Id from Floc userId module and passing it to PubMatic SSP in user.eids and user.data.segment

### CODE REVIEW

**Comment by patmmccann** [comment]
does it make sense to write it to those places in the prebid fpd object rather than just the one that goes to your endpoint? It sseems this work is nearly 100% likely to get duplicated shortly

**Comment by pm-shashank-jain** [comment]
> does it make sense to write it to those places in the prebid fpd object rather than just the one that goes to your endpoint? It sseems this work is nearly 100% likely to get duplicated shortly

@patmmccann can you please elaborate, didn't understand it!
If it helps, let me just explain what I am doing is, reading the FloC ID from userId in validBids passed to adapter and passing it in user.data.segment and user.eids to PubMatic SSP. Doing this until FloC data position becomes standard i.e. either in eids or user data-segments or some place else.

**Comment by patmmccann** [comment]
I am suggesting that instead of copying it there for your endpoint payload, you copy it there for everyone

**Comment by abhinavsinha001** [comment]
@patmmccann  As per https://github.com/prebid/Prebid.js/issues/6510, I believe there is no consensus on  where FloC id should end up in oRTB and for the current origin trial adapters can choose where to pass this ID.
Hence we decided not to add it to common object as of now, and hopefully move the code to common FPD ortb2 object once we have location finalised.

**Comment by patmmccann** [comment]
Fair enough

File: modules/pubmaticBidAdapter.js [hunk 1]
```diff
@@ -708,8 +713,67 @@ function _addFloorFromFloorModule(impObj, bid) {
...
+      payload.user.data = [];
+    }
+    payload.user.data.push(flocObject);
+  }
+}
+
 function _handleEids(payload, validBidRequests) {
-  const bidUserIdAsEids = utils.deepAccess(validBidRequests, '0.userIdAsEids');
+  var bidUserIdAsEids = utils.deepAccess(validBidRequests, '0.userIdAsEids');
+  var flocObject = _getFlocId(validBidRequests, FLOC_FORMAT.EID);
```
Comment (musikele) [inline modules/pubmaticBidAdapter.js:770]: consider using `let` instead

**Review by musikele — COMMENTED** [review]
just a very small note

File: modules/pubmaticBidAdapter.js (hunk 1 above)
Comment (pm-shashank-jain) [reply modules/pubmaticBidAdapter.js]: Done

**Review by musikele — APPROVED** [review]
LGTM

**Pull request merged**